jung-api/src/main/java/edu/uci/ics/jung/graph/DelegateCTreeNetwork.java

```diff
@@ -194,7 +194,8 @@ private void setDepth(N node, N parent) {
       height = Optional.of(0);
     } else {
       int nodeDepth = depths.get(parent) + 1;
-      height = Optional.of(Math.max(nodeDepth, height.get()));
+      height =
+          Optional.of(Math.max(nodeDepth, (height.equals(Optional.empty()) ? 0 : height.get())));
```

Comment:
I'm going to try editing this change in place to clean up the syntax; let me know how it looks from your side.